ui/management/commands/sync_video_key_with_edx.py

```diff
@@ -60,9 +61,23 @@ def handle(self, *args, **options):
                     )
                     self.stdout.write(self.style.ERROR(str(exc)))
 
-            for vid in course_videos:
-                Video.objects.filter(title=vid.get("client_video_id")).update(
-                    key=vid.get("edx_video_id")
+            latest_videos_by_created_at = {}
+            for video in course_videos:
+                key = video.get("encoded_videos", [{}])[0].get("url", "").split("/")[-2]
+                created = datetime.fromisoformat(video.get("created", "1970-01-01"))
+                if (
+                    key and key not in latest_videos_by_created_at
+                ) or created > datetime.fromisoformat(
+                    latest_videos_by_created_at[key].get("created", "1970-01-01")
+                ):
```

Comment:
Same here. If we get `key=""`, it will raise a `KeyError`. This should be as:
```suggestion
                if key and (
                    key not in latest_videos_by_created_at
                 or created > datetime.fromisoformat(
                    latest_videos_by_created_at[key].get("created", "1970-01-01")
                )):
```